### ISSUE TITLE
[ray] Fix auto batch size and learning rate detection with remote backend

### ISSUE THREAD

**Issue by tgaddair** [ludwig-ai/ludwig#1409]
In the current implementation, both the `tune_learning_rate` and `tune_batch_size` methods of the Trainer are not implemented for remote backends. This means they will default to running locally, but this is not useful if the remote trainer uses a GPU, as the head node will not have a GPU in its context.

Similarly, the current implementation sets the batch_size and learning_rate properties on the Trainer itself, instead of returning them as a pure function. This means that the remote trainer will not be able to pick up the new batch size, as it will be running a different instance of the trainer in the remote context.

What we should do instead is change these functions to run remotely, with GPUs if available. We should also remove side effects from these functions so that the learning rate and batch size can be passed in to the remote trainer statelessly.
